### PR TITLE
Changed error message

### PR DESCRIPTION
**Pull request by in4margaret**

### CODE REVIEW

**Comment by in4margaret** [comment]
Maybe we need to update all errors and move them to a separate file.

**Comment by rozele** [comment]
Agree - let's fix the rest of these in this PR. We could make the ctors internal and add the configuration validation logic to the service factories.

File: src/NLU.DevOps.Luis/LuisClient.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ internal class LuisClient : ILuisClient
             {
                 if (authoringKey == null || authoringRegion == null)
                 {
-                    throw new InvalidOperationException("Must provide authoring key and region to perform authoring operations.");
+                    throw new InvalidOperationException("Must provide authoring key and region to perform authoring operations. See: https://microsoft.github.io/NLU.DevOps/docs/LuisSecrets.html#luisauthoringkey .");
```
Comment (rozele) [inline src/NLU.DevOps.Luis/LuisClient.cs]: Not convinced we want to include a specific GitHub URL here.

File: src/NLU.DevOps.Luis/LuisNLUService.cs [hunk 2]
```diff
@@ -243,9 +243,10 @@ private async Task PollTrainingStatusAsync(CancellationToken cancellationToken)
 
                 if (!inProgress)
                 {
-                    if (trainingStatus.Any(modelInfo => modelInfo.Details.Status == "Fail"))
+                    var failed = trainingStatus.Where(modelInfo => modelInfo.Details.Status == "Fail").FirstOrDefault();
+                    if (failed != null)
                     {
-                        throw new InvalidOperationException("Failure occurred while training LUIS model.");
+                        throw new InvalidOperationException($"Failure occurred while training LUIS model. {failed.Details.FailureReason}");
```
Comment (rozele) [inline src/NLU.DevOps.Luis/LuisNLUService.cs]: Insert `Reason: {failed.Details...}`

**Comment by rozele** [comment]
I think we need similar changes for Lex.

**Pull request closed without merging**